fdbserver/RestoreMaster.actor.cpp

```diff
@@ -157,6 +164,28 @@ ACTOR Future<Void> startProcessRestoreRequests(Reference<RestoreMasterData> self
 
 	// lock DB for restore
 	wait(lockDatabase(cx, randomUID));
+	state int numTries = 0;
+	loop {
+		try {
+			wait(lockDatabase(cx, randomUID));
```

Comment:
I'm confused by the need for this loop overall?  You're locking the database above the loop, and then in the loop, and then checking to see that it's locked.  This sounds like lockDatabase isn't actually locking the database then?  What problem were you running into?

Reply:
It's my copy-paste fault. I intend to replace the `wait(lockDatabase(...))` with the loop. 

We want to lock DB with the `randomUID`. If the DB is currently locked with a different UID, the `lockDatabase(..)` will throw errors. We should wait until the DB is unlocked so that we can properly lock the DB. 

I will remove the `lockDatabase(...)` outside the `loop`.